### ISSUE TITLE
Gracefully handle non-amd files

### ISSUE THREAD

**Issue by stevage** [tanem/recast-deamdify#4]
This file exploded for me:

```
var profile = {
    resourceTags : {
        amd : function(filename, mid) {
            "use strict";
            return (/\.js$/).test(filename);
        }
    }
};
```

Obviously it's not AMD, but it's one file in a huge codebase that is 99% AMD, so it'd be handy if it could just gracefully copy it over without exploding.

**Comment by stevage** [comment]
Hmm, seems to be fixed by moving the line

`      var factoryBody = getFactoryBody(body);`

to just before the line

`      ast.program.body = transform(factoryBody);`

in deamdify.js

**Comment by tanem** [comment]
Hey @stevage, thanks for the heads up!

Your proposal makes sense, I've implemented it in the referenced commit, using your code as a new test fixture. Do you want to cast your eye over it to make sure it's working for you before I merge it down to `master`?

**Comment by stevage** [comment]
Looks good to me!

**Comment by tanem** [comment]
Published as `v1.1.4` :v:
